lib/collection.js

```diff
@@ -1239,14 +1240,16 @@ Collection.prototype.createIndex = function(fieldOrSpec, options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
   options = options || {};
 
-  const createIndexOperation = new CreateIndexOperation(
-    this.s.db,
+  const prepared = prepareCreateIndex(fieldOrSpec, options);
+  const createIndexesOperation = new CreateIndexesOperation(
+    this,
```

Comment:
My understanding was the parent should be the collection for methods in `Collection` and the db for methods in `Db`, so I changed this from `this.s.db` to `this`. I'm not 100% sure this is correct, so LMK if it's wrong and I'll change back to `this.s.db`, here and also in `Collection.createIndexes`.

Reply:
yes it should always be `this`, we're using the `parent` to resolve the destination of the command